src/warc2zim/content_rewriting/js.py

```diff
@@ -111,128 +178,147 @@ def f(m_object, opts):
-        (re.compile(r"[=,]\s*\bthis\b\s*(?![\s\w:.$])"), replaceThis()),
-        # rewrite `})(thisRw)`
-        (re.compile(r"\}(?:\s*\))?\s*\(this\)"), replaceThis()),
+        (re.compile(r"[=,]\s*\bthis\b\s*(?![\s\w:.$])"), replace_this()),
+        # rewrite `})(this_rw)`
+        (re.compile(r"\}(?:\s*\))?\s*\(this\)"), replace_this()),
         # rewrite this in && or || expr
-        (re.compile(r"[^|&][|&]{2}\s*this\b\s*(?![|\s&.$](?:[^|&]|$))"), replaceThis()),
-        # ignore `async import`, custom function
+        (
+            re.compile(r"[^|&][|&]{2}\s*this\b\s*(?![|\s&.$](?:[^|&]|$))"),
+            replace_this(),
+        ),
+        # ignore `async import`.
+        # As the rule will match first, it will prevent next rule matching `import` to be apply
+        # to `async import`.
         (re.compile(r"async\s+import\s*\("), m2str(lambda x: x)),
         # esm dynamic import, if found, mark as module
         (
             re.compile(r"[^$.]\bimport\s*\("),
-            replaceImport("import", "____wb_rewrite_import__"),
+            replace_import("import", "____wb_rewrite_import__"),
         ),
     ]
 
 
 class JsRewriter:
+    """
+    JsRewriter is in charge of rewriting the js code stored in our zim file.
+
+    The main "input" is a list of rules, each rule being a tuple (regex, rewriting_function).
+    We want to applay each rule to the content. But doing it blindly is conter productives.
+    It would means that we have to do N replacements (N == number of rules).
+    To avoid that, we create one unique regex (`compiled_rule`) equivalent to
+    `(regex0|regex1|regex2|...)` and we do only one replacement with this regex.
+    When we have a match, we do N regex search to know which rules is corresponding and we apply
+    the asosociated rewriting_function.
+    """
+
     def __init__(
         self,
         url_rewriter: Callable[[str], str],
-        extraRules: Optional[Iterable[TransformationRule]] = None,
+        extra_rules: Optional[Iterable[TransformationRule]] = None,
     ):
-        self.extraRules = extraRules
-        self.firstBuff = self.initLocalDeclaration(GLOBAL_OVERRIDES)
-        self.lastBuff = "\n\n}"
+        self.extra_rules = extra_rules or []
+        self.first_buff = self._init_local_declaration(GLOBAL_OVERRIDES)
+        self.last_buff = "\n\n}"
         self.url_rewriter = url_rewriter
 
-    def initLocalDeclaration(self, localDecls: Iterable[str]) -> str:
-        ASSIGN_FUNC = "_____WB$wombat$assign$function_____"
-        buffer = f"""var {ASSIGN_FUNC} = function(name) {{return (self._wb_wombat && self._wb_wombat.local_init && self._wb_wombat.local_init(name)) || self[name]; }};
-if (!self.__WB_pmw) {{ self.__WB_pmw = function(obj) {{ this.__WB_source = obj; return this; }} }}
-{{
-"""
-        for decl in localDecls:
-            buffer += f"""let {decl} = {ASSIGN_FUNC}("{decl}");\n"""
+    def _init_local_declaration(self, local_decls: Iterable[str]) -> str:
```

Comment:
Maybe a docstring stating that we always start with this boilerplate code

Reply:
Indeed. Even more as we don't always start with this boilerplate code.